test/plugins/net.spec.js

```diff
@@ -165,5 +163,87 @@ describe('Plugin', () => {
         })
       })
     })
+
+    it('should instrument close', done => {
+      const socket = new net.Socket()
+
+      agent
+        .use(traces => {
+          expect(traces[0][0]).to.deep.include({
+            name: 'tcp.connect',
+            service: 'test-tcp',
+            resource: `localhost:${port}`
+          })
+          expect(traces[0][0].meta).to.deep.include({
+            'span.kind': 'client',
+            'tcp.family': 'IPv4',
+            'tcp.remote.host': 'localhost',
+            'tcp.remote.port': `${port}`,
+            'out.host': 'localhost',
+            'out.port': `${port}`
+          })
+          expect(traces[0][0].parent_id.toString()).to.equal(parent.context().toSpanId())
+        })
+        .then(done)
+        .catch(done)
+
+      tracer.scope().activate(parent, () => {
+        socket.connect({ port })
+        socket.destroy()
```

Comment:
I removed the test completely since it would require an external firewall to explicitly drop the TCP SYN packet, similar to why the `timeout` event is not tested. Since the list of events is defined as an array in code, testing one should be sufficient to know that they all work.